src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -2794,8 +2794,8 @@ exprt c_typecheck_baset::do_special_functions(
     return nil_exprt();
 }
 
-/// \param type:checked arguments, type-checked function
-/// \return type-adjusted function arguments
+/// Typecheck the parameters in a function call expression, inserting
+/// implicit casts around parameters where necessary.
```

Comment:
> inserting implicit casts around parameters

:open_mouth: But _why_? I get that you're just documenting the existing behaviour, but I'm really not sure why a function called _check_ actually ends up modifying the expression it's supposedly checking...

Reply:
I think in CBMC we use 'typecheck' to mean a bit more than what a typical compiler would mean by 'typecheck'... It seems to be a sort of 'typecheck and canonicalise' operation - which, while I agree the naming is not great, is not entirely an unreasonable thing to do.

Reply:
Just as a comment on the proposed new wording: we insert casts, which in fact makes them no longer _implicit_.